From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Martijn Muijsers <martijnmuijsers@live.nl>
Date: Mon, 26 Dec 2022 00:17:40 +0100
Subject: [PATCH] Make logging login locations configurable

License: GPL-3.0-only (https://www.gnu.org/licenses/gpl-3.0.html)
Gale - https://galemc.org

This patch is based on the following patch:
"configurable logging of player login location"
By: Simon Gardling <titaniumtown@gmail.com>
As part of: JettPack (https://gitlab.com/Titaniumtown/JettPack)
Licensed under: GPL-3.0-only (https://www.gnu.org/licenses/gpl-3.0.html)

diff --git a/net/minecraft/server/players/PlayerList.java b/net/minecraft/server/players/PlayerList.java
index 59b42c8676c26d40c8717ce798beb062e53272ad..6993d9da8e05248a8f1d628f9d534c816ced293d 100644
--- a/net/minecraft/server/players/PlayerList.java
+++ b/net/minecraft/server/players/PlayerList.java
@@ -317,6 +317,7 @@ public abstract class PlayerList {
         }
         // Paper end - Configurable player collision
         org.purpurmc.purpur.task.BossBarTask.addToAll(player); // Purpur - Implement TPSBar
+        if (org.galemc.gale.configuration.GaleGlobalConfiguration.get().logToConsole.playerLoginLocations) { // Gale - JettPack - make logging login location configurable
         // CraftBukkit start - moved down
         LOGGER.info(
             "{}[{}] logged in with entity id {} at ([{}]{}, {}, {})", // Paper - add world identifier
@@ -329,6 +330,16 @@ public abstract class PlayerList {
             player.getZ()
         );
         // CraftBukkit end - moved down
+            // Gale start - JettPack - make logging login location configurable
+        } else {
+            LOGGER.info(
+                    "{}[{}] logged in with entity id {}", // CraftBukkit - add world name
+                    player.getPlainTextName(),
+                    address,
+                    player.getId()
+            );
+        }
+        // Gale end - JettPack - make logging login location configurable
         // Paper start - Send empty chunk, so players aren't stuck in the world loading screen with our chunk system not sending chunks when dead
         if (player.isDeadOrDying()) {
             net.minecraft.core.Holder<net.minecraft.world.level.biome.Biome> plains = level.registryAccess().lookupOrThrow(net.minecraft.core.registries.Registries.BIOME)
